TrackingTools/DetLayers/interface/rangesIntersect.h

```diff
@@ -17,7 +17,7 @@ inline bool rangesIntersect(const Range& a, const Range& b) {
 
 template <typename Range, typename Less>
 inline bool rangesIntersect(const Range& a, const Range& b, Less const& less) {
-  return !(less(b.second, a.first) | less(a.second, b.first));
+  return !(less(b.second, a.first) || less(a.second, b.first));
```

Comment:
here and in a few other places the bitwise logic could/should stay.
Note that for some reason in this file only one method is changed.